pep/README.md

```diff
@@ -0,0 +1,166 @@
+# Policy Enforcement Point (PEP)
+
+NGINX-based reference implementation that enforces authorization decisions from Knooppunt's PDP.
+
+## Architecture
+
+```
+┌─────────────────┐
+│ External Client │
+└────────┬────────┘
+         │ GET /fhir/Patient/123
+         │ Authorization: Bearer <token>
+         ▼
+┌────────────────────────────────────────┐
+│           PEP (NGINX)                  │
+│  ┌──────────────────────────────────┐  │
+│  │ 1. Extract token                 │  │
+│  │ 2. Introspect (mock for testing) │  │
+│  │ 3. Build OPA request             │  │
+│  │ 4. Call PDP for decision         │  │
+│  │ 5. Enforce (allow/deny)          │  │
+│  └──────────────────────────────────┘  │
+└────────┬─────────────────┬─────────────┘
+         │                 │
+         │ POST /v1/data/  │
+         │  knooppunt/authz│
+         ▼                 │
+┌────────────────────┐     │
+│ Knooppunt PDP      │     │
+│ (port 8081)        │     │
+│ - Validates input  │     │
+│ - Returns allow/deny     │
+└────────────────────┘     │
+                           │ (if allowed)
+                           ▼
+                  ┌────────────────┐
+                  │  FHIR Server   │
+                  └────────────────┘
+```
+
+## Quick Start
+
+```bash
+# Start PEP with rest of stack
+docker compose --profile pep up -d
+
+# Test with mock token (format: bearer-<ura>-<uzi_role>-<practitioner_id>-<bsn>)
+curl -H "Authorization: Bearer bearer-00000020-01.015-123456789-900186021" \
+  http://localhost:9080/fhir/Patient/patient-123
+```
+
+**Endpoints:**
+- PEP: `http://localhost:9080`
+- PDP: `http://localhost:8081/v1/data/knooppunt/authz` (internal API)
+
+## How It Works
+
+1. Extract bearer token from `Authorization` header
+2. Introspect token via `/_introspect` endpoint (mock NJS function for testing)
+3. Extract FHIR context (resource type, ID) from request URI
+4. Build OPA request and call Knooppunt PDP
+5. Enforce decision: allow (200) or deny (403)
+
+## Configuration
+
+Environment variables in `docker-compose.yml`:
+
+```yaml
+# Backend connections
+FHIR_BACKEND_HOST=hapi-fhir   # FHIR server
+FHIR_BACKEND_PORT=7050
+KNOOPPUNT_PDP_HOST=knooppunt  # PDP endpoint
+KNOOPPUNT_PDP_PORT=8081       # Internal API only
+
+# Data holder configuration (organization where data is stored)
+DATA_HOLDER_ORGANIZATION_URA=00000666
+DATA_HOLDER_FACILITY_TYPE=Z3
+
+# Request configuration
+REQUESTING_FACILITY_TYPE=Z3
+PURPOSE_OF_USE=TREAT
+EVENT_CODE=GGC002
+```
+
+## OPA Request Format
+
+The PEP sends requests with clear field names matching XACML/Mitz terminology:
+
+```json
+POST /v1/data/knooppunt/authz
```

Comment:
I think this should be prefixed with `/pdp` as the base path to match our component structure.